presto-hive/src/main/java/io/prestosql/plugin/hive/s3/PrestoS3FileSystem.java

```diff
@@ -1255,6 +1262,14 @@ private void uploadObject()
                             break;
                     }
                 }
+                switch (s3StorageClass) {
+                    case STANDARD:
+                        request.withStorageClass(Standard);
+                        break;
+                    case IntelligentTiering:
+                        request.withStorageClass(IntelligentTiering);
```

Comment:
It would be better to encode the mapping from our enum to AWS SDK enum in our enum directly:

```
public enum S3StorageClass
{
    STANDARD(StorageClass.Standard),
    INTELLIGENT_TIERING(StorageClass.IntelligentTiering),
    /**/;

    private StorageClass awsStorageClass;

    S3StorageClass(StorageClass awsStorageClass)
    {
        this.awsStorageClass = requireNonNull(awsStorageClass, "awsStorageClass is null");
    }

    public StorageClass toAwsStorageClass()
    {
        return awsStorageClass;
    }
}
```

Then you would just call `toAwsStorageClass()` here, instead of a switch.

Reply:
@electrum do you have a better field name idea than `awsStorageClass`?

Reply:
Let's keep the class name `PrestoS3StorageClass`, as this is consistent with the other S3 enums. Then we can name the field `storageClass` and method `toStorageClass()`.

`PrestoS3AclType` is another place where we do this type of mapping.

Reply:
@findepi used `getS3StorageClass` instead of `toAwsStorageClass`, looked consistent with acltype class example you have given I hope that's okay.